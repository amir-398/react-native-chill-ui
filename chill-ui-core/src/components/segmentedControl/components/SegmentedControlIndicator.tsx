import { Animated } from 'react-native';
import { useEffect, useRef } from 'react';
import { AnimatedBox } from '@components/animatedBox';
import { SegmentedControlIndicatorPropsTw } from '@types';
import { cn, classNameHandler, classNamePropsHandler, styleHandler } from '@utils';

import { styles } from '../styles/SegmentedControl.ss.styles';
import { twStyles } from '../styles/SegmentedControl.tw.styles';
import { segmentedControlDefaultProps } from '../utils/defaultProps';
import { useSegmentedControlState } from '../context/SegmentedControlContext';

/**
 * Animated visual indicator for the selected option.
 *
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * <SegmentedControlIndicator />
 * ```
 *
 * @param className - Custom CSS classes for styling the indicator (NativeWind)
 * @param duration - Animation duration in milliseconds for position transitions (default: 200)
 * @param style - Style object for the indicator (React Native)
 * @returns SegmentedControlIndicator component with animated positioning
 * @throws Error if used outside of SegmentedControlProvider context
 */
export function SegmentedControlIndicator(props: SegmentedControlIndicatorPropsTw) {
  classNamePropsHandler(props, 'SegmentedControlIndicator');
  const { itemWidth, selectedOption, validItemsValues } = useSegmentedControlState();
  const { className, duration, style } = props;

  const animatedLeft = useRef(
    new Animated.Value(
      itemWidth * validItemsValues.indexOf(selectedOption) + segmentedControlDefaultProps.internalPadding / 2,
    ),
  ).current;

  useEffect(() => {
    const newLeft =
      itemWidth * validItemsValues.indexOf(selectedOption) + segmentedControlDefaultProps.internalPadding / 2;
    Animated.timing(animatedLeft, {
      duration: duration ?? segmentedControlDefaultProps.duration,
      toValue: newLeft,
      useNativeDriver: false,
    }).start();
  }, [selectedOption, validItemsValues, itemWidth, animatedLeft, duration]);

  return (
    <AnimatedBox
      {...classNameHandler(cn(twStyles.indicatorContainer, className, twStyles.indicatorContainerFreezed))}
      {...styleHandler({
        defaultStyle: [
          styles.indicatorContainer,
          styles.indicatorContainerFreezed,
          {
            left: animatedLeft,
            width: itemWidth,
          },
        ],
        style,
      })}
    />
  );
}

SegmentedControlIndicator.displayName = 'SegmentedControlIndicator';
