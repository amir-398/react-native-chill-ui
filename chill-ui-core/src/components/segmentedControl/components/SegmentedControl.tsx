import { PropsWithChildren, useMemo } from 'react';

import { SegmentedControlProvider } from './SegmentedControlProvider';
import { SegmentedControlIndicator } from './SegmentedControlIndicator';
import { SegmentedControlPanelContent } from './SegmentedControlPanelContent';
import { SegmentedControlTriggerContent } from './SegmentedControlTriggerContent';
import { SegmentedControlPanelSliderContent } from './SegmentedControlPanelSliderContent';

/**
 * Tab-like interface for switching between options.
 *
 * @example
 * ```tsx
 * <SegmentedControl>
 *   <SegmentedControlTriggerContent>
 *     <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
 *     <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
 *   </SegmentedControlTriggerContent>
 *   <SegmentedControlIndicator />
 *   <SegmentedControlPanelContent>
 *     <SegmentedControlPanel value="option1">
 *       <Box>Content for Option 1</Box>
 *     </SegmentedControlPanel>
 *     <SegmentedControlPanel value="option2">
 *       <Box>Content for Option 2</Box>
 *     </SegmentedControlPanel>
 *   </SegmentedControlPanelContent>
 * </SegmentedControl>
 * ```
 *
 * @param children - SegmentedControlTriggerContent, SegmentedControlIndicator, and SegmentedControlPanelContent components (required)
 * @returns SegmentedControl component that manages segmented control state and context
 */
export function SegmentedControl(props: PropsWithChildren<any>) {
  const { children } = props;

  const childrenArray = useMemo(() => {
    if (!children) return [];
    return Array.isArray(children) ? children : [children];
  }, [children]);

  const contentChildren = useMemo(
    () => childrenArray.filter(item => item.type === SegmentedControlTriggerContent),
    [childrenArray],
  );

  const panelsChildren = useMemo(
    () =>
      childrenArray.filter(
        item => item.type === SegmentedControlPanelContent || item.type === SegmentedControlPanelSliderContent,
      ),
    [childrenArray],
  );

  const indicatorChildren = useMemo(
    () => childrenArray.filter(item => item.type === SegmentedControlIndicator),
    [childrenArray],
  );

  return (
    <SegmentedControlProvider indicatorChildren={indicatorChildren}>
      {contentChildren}
      {panelsChildren}
    </SegmentedControlProvider>
  );
}

SegmentedControl.displayName = 'SegmentedControl';
