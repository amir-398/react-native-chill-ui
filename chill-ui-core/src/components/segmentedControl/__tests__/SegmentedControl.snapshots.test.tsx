import { Text, View } from 'react-native';
import { render } from '@testing-library/react-native';

import { SegmentedControl } from '../components/SegmentedControl';
import { SegmentedControlPanel } from '../components/SegmentedControlPanel';
import { SegmentedControlTrigger } from '../components/SegmentedControlTrigger';
import { SegmentedControlIndicator } from '../components/SegmentedControlIndicator';
import { SegmentedControlPanelContent } from '../components/SegmentedControlPanelContent';
import { SegmentedControlTriggerContent } from '../components/SegmentedControlTriggerContent';

// Mock utils
jest.mock('../../../utils', () => ({
  classNameHandler: jest.fn(() => ({})),
  classNamePropsHandler: jest.fn(),
  cn: jest.fn((...args) => args.filter(Boolean).join(' ')),
  isString: jest.fn(value => typeof value === 'string'),
  SlotTw: ({ children }: any) => children,
  styleHandler: jest.fn(() => ({})),
}));

// Mock AnimatedBox
jest.mock('../../../components/animatedBox', () => ({
  AnimatedBox: ({ children, ...props }: any) => {
    const { View: ReactNativeView } = require('react-native');
    return (
      <ReactNativeView testID="animated-box" {...props}>
        {children}
      </ReactNativeView>
    );
  },
}));

// Mock Box
jest.mock('../../../components/box', () => ({
  Box: ({ children, ...props }: any) => {
    const { View: ReactNativeView } = require('react-native');
    return (
      <ReactNativeView testID="box" {...props}>
        {children}
      </ReactNativeView>
    );
  },
}));

// Mock String
jest.mock('../../../components/string', () => ({
  String: ({ children, ...props }: any) => {
    const { Text: ReactNativeText } = require('react-native');
    return (
      <ReactNativeText testID="string" {...props}>
        {children}
      </ReactNativeText>
    );
  },
}));

// Mock Animated
jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');
  RN.Animated.timing = () => ({
    start: () => {},
  });
  return RN;
});

describe('SegmentedControl Snapshots', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot with basic props', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent>
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with default value', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent defaultValue="option2">
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom styling', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent className="custom-content" style={{ padding: 8 }}>
          <SegmentedControlTrigger
            value="option1"
            className="custom-trigger"
            activeClassName="active-trigger"
            style={{ margin: 4 }}
            activeStyle={{ backgroundColor: '#3B82F6' }}
          >
            Option 1
          </SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator className="custom-indicator" style={{ height: 6 }} />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with disabled state', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent>
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2" isDisabled>
            Option 2
          </SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with panels content', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent defaultValue="option1">
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
        <SegmentedControlPanelContent>
          <SegmentedControlPanel value="option1">
            <View style={{ backgroundColor: '#DBEAFE', borderRadius: 8, padding: 16 }}>
              <Text style={{ color: '#1E40AF', fontSize: 16, fontWeight: 'bold' }}>Content for Option 1</Text>
            </View>
          </SegmentedControlPanel>
          <SegmentedControlPanel value="option2">
            <View style={{ backgroundColor: '#FEE2E2', borderRadius: 8, padding: 16 }}>
              <Text style={{ color: '#991B1B', fontSize: 16, fontWeight: 'bold' }}>Content for Option 2</Text>
            </View>
          </SegmentedControlPanel>
        </SegmentedControlPanelContent>
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom duration', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent>
          <SegmentedControlTrigger value="option1">Fast Animation</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator duration={100} />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom internal padding', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent internalPadding={16}>
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with touchable-opacity', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent>
          <SegmentedControlTrigger value="option1" as="touchable-opacity">
            Touchable Option
          </SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with multiple triggers', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent>
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option3">Option 3</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option4">Option 4</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with forceRender prop', () => {
    const tree = render(
      <SegmentedControl>
        <SegmentedControlTriggerContent defaultValue="option1">
          <SegmentedControlTrigger value="option1">Option 1</SegmentedControlTrigger>
          <SegmentedControlTrigger value="option2">Option 2</SegmentedControlTrigger>
        </SegmentedControlTriggerContent>
        <SegmentedControlIndicator />
        <SegmentedControlPanelContent>
          <SegmentedControlPanel value="option1">
            <Text>Content 1</Text>
          </SegmentedControlPanel>
          <SegmentedControlPanel value="option2" forceRender>
            <Text>Content 2 (Force Rendered)</Text>
          </SegmentedControlPanel>
        </SegmentedControlPanelContent>
      </SegmentedControl>,
    );
    expect(tree).toMatchSnapshot();
  });
});
