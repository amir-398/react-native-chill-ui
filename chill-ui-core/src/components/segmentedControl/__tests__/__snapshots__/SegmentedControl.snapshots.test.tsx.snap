// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl Snapshots should match snapshot with basic props 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 2
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with custom duration 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Fast Animation
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with custom internal padding 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 2
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with custom styling 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        {
          "margin": 4,
        },
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 1
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with default value 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 2
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with disabled state 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 2
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with forceRender prop 1`] = `
[
  <View
    onLayout={[Function]}
    testID="box"
  >
    <View
      testID="animated-box"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "width": -Infinity,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            false,
            undefined,
          ]
        }
        testID="string"
      >
        Option 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "width": -Infinity,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            false,
            undefined,
          ]
        }
        testID="string"
      >
        Option 2
      </Text>
    </View>
  </View>,
  <View
    testID="box"
  >
    <View
      testID="box"
    >
      <Text>
        Content 2 (Force Rendered)
      </Text>
    </View>
  </View>,
]
`;

exports[`SegmentedControl Snapshots should match snapshot with multiple triggers 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 2
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 3
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "width": -Infinity,
        },
        false,
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Option 4
    </Text>
  </View>
</View>
`;

exports[`SegmentedControl Snapshots should match snapshot with panels content 1`] = `
[
  <View
    onLayout={[Function]}
    testID="box"
  >
    <View
      testID="animated-box"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "width": -Infinity,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            false,
            undefined,
          ]
        }
        testID="string"
      >
        Option 1
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "width": -Infinity,
          },
          false,
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            false,
            undefined,
          ]
        }
        testID="string"
      >
        Option 2
      </Text>
    </View>
  </View>,
  <View
    testID="box"
  />,
]
`;

exports[`SegmentedControl Snapshots should match snapshot with touchable-opacity 1`] = `
<View
  onLayout={[Function]}
  testID="box"
>
  <View
    testID="animated-box"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "width": -Infinity,
      }
    }
  >
    <Text
      style={
        [
          false,
          undefined,
        ]
      }
      testID="string"
    >
      Touchable Option
    </Text>
  </View>
</View>
`;
